### ISSUE TITLE
Flutter framework RichText crash when Accessibility Service is enabled

### ISSUE THREAD

**Issue by singun-lxd** [flutter/flutter#65818]
I've found a RichText crash when Accessibility Service is enabled

Flutter version 1.17.5 stable

## Steps to Reproduce

1. Create a demo project
2. Write code as follows:

```
class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'RichText Crash Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: Scaffold(
        backgroundColor: Colors.white,
        body: Center(
          child: RichText(
            text: TextSpan(children: [
              // remove [WidgetSpan] and crash not happen any more
              WidgetSpan(child: SizedBox.shrink()),
              TextSpan(
                text: "Text",
                style: TextStyle(color: Colors.black),
                // remove [recognizer] and crash not happen any more
                recognizer: TapGestureRecognizer()..onTap = () {},
              )
            ]),
          ),
        ),
      ),
    );
  }
}
```

3. Run the code above and enable **ONE OF** the Accessibility Services in your system
4. When WidgetSpan is removed **OR** recognizer in TextSpan is removed, the crash not happen
5. Full demo in github: https://github.com/singun-lxd/flutter_richtext_crash_demo

**Expected results:**
nothing crash

**Actual results:** 
Dart code crashes, you can see crash stace traces below in 'Logs'

## Temp Solution

To avoid crash, add these two lines **AFTER** `WidgetsFlutterBinding.ensureInitialized()` and **BEFORE** `runApp`
```
window.onSemanticsEnabledChanged = () {};
RendererBinding.instance.setSemanticsEnabled(false);
```
This code is used to disable Accessibility Service on Flutter

<details>
  <summary>Logs</summary>

```
I/flutter (13840): ══╡ EXCEPTION CAUGHT BY SCHEDULER LIBRARY ╞═════════════════════════════════════════════════════════
I/flutter (13840): The following RangeError was thrown during a scheduler callback:
I/flutter (13840): RangeError (index): Invalid value: Valid value range is empty: 0
I/flutter (13840): 
I/flutter (13840): When the exception was thrown, this was the stack:
I/flutter (13840): #0      List.[] (dart:core-patch/growable_array.dart:146:60)
I/flutter (13840): #1      List.elementAt (dart:core-patch/growable_array.dart:355:16)
I/flutter (13840): #2      RenderParagraph.assembleSemanticsNode (package:flutter/src/rendering/paragraph.dart:899:50)
I/flutter (13840): #3      _SwitchableSemanticsFragment.compileChildren (package:flutter/src/rendering/object.dart:3684:13)
I/flutter (13840): #4      _SyncIterator.moveNext (dart:core-patch/core_patch.dart:146:23)
I/flutter (13840): #5      ExpandIterator.moveNext (dart:_internal/iterable.dart:478:31)
I/flutter (13840): #6      new List.from (dart:core-patch/array_patch.dart:47:19)
I/flutter (13840): #7      Iterable.toList (dart:core/iterable.dart:400:12)
I/flutter (13840): #8      _SwitchableSemanticsFragment.compileChildren (package:flutter/src/rendering/object.dart:3681:8)
I/flutter (13840): #9      _SyncIterator.moveNext (dart:core-patch/core_patch.dart:146:23)
I/flutter (13840): #10     _SyncIterator.moveNext (dart:core-patch/core_patch.dart:139:32)
I/flutter (13840): #11     ExpandIterator.moveNext (dart:_internal/iterable.dart:478:31)
I/flutter (13840): #12     new List.from (dart:core-patch/array_patch.dart:47:19)
I/flutter (13840): #13     Iterable.toList (dart:core/iterable.dart:400:12)
I/flutter (13840): #14     _SwitchableSemanticsFragment.compileChildren (package:flutter/src/rendering/object.dart:3681:8)
I/flutter (13840): #15     _SyncIterator.moveNext (dart:core-patch/core_patch.dart:146:23)
I/flutter (13840): #16     ExpandIterator.moveNext (dart:_internal/iterable.dart:478:31)
I/flutter (13840): #17     new List.from (dart:core-patch/array_patch.dart:47:19)
I/flutter (13840): #18     Iterable.toList (dart:core/iterable.dart:400:12)
I/flutter (13840): #19     _RootSemanticsFragment.compileChildren (package:flutter/src/rendering/object.dart:3562:8)
I/flutter (13840): #20     _SyncIterator.moveNext (dart:core-patch/core_patch.dart:146:23)
I/flutter (13840): #21     Iterable.single (dart:core/iterable.dart:554:13)
I/flutter (13840): #22     RenderObject._updateSemantics (package:flutter/src/rendering/object.dart:2584:7)
I/flutter (13840): #23     PipelineOwner.flushSemantics (package:flutter/src/rendering/object.dart:1083:16)
I/flutter (13840): #24     RendererBinding.drawFrame (package:flutter/src/rendering/binding.dart:407:21)
I/flutter (13840): #25     WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:884:13)
I/flutter (13840): #26     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:284:5)
I/flutter (13840): #27     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1113:15)
I/flutter (13840): #28     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1052:9)
I/flutter (13840): #29     SchedulerBinding.scheduleWarmUpFrame.<anonymous closure> (package:flutter/src/scheduler/binding.dart:861:7)
I/flutter (13840): (elided 11 frames from class _RawReceivePortImpl, class _Timer, dart:async, and dart:async-patch)
I/flutter (13840): ════════════════════════════════════════════════════════════════════════════════════════════════════
I/flutter (13840): Another exception was thrown: 'package:flutter/src/widgets/binding.dart': Failed assertion: line 863 pos 14: '!_firstFrameCompleter.isCompleted': is not true.

```

flutter doctor -v
```
[✓] Flutter (Channel stable, v1.17.5, on Mac OS X 10.15.6 19G2021, locale zh-Hans-CN)
    • Flutter version 1.17.5 at /Users/singun/Library/Flutter/sdk
    • Framework revision 8af6b2f038 (3 months ago), 2020-06-30 12:53:55 -0700
    • Engine revision ee76268252
    • Dart version 2.8.4

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/singun/Library/Android/sdk
    • Platform android-29, build-tools 29.0.2
    • ANDROID_HOME = /Users/singun/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.7)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.7, Build version 11E801a
    • CocoaPods version 1.8.4

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 49.0.2
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.27.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 2.21.1

[✓] Connected device (1 available)
    • JSN L22 • NQX4C20408006356 • android-arm64 • Android 9 (API 28)

• No issues found!
```

</details>

**Comment by darshankawar** [comment]
Issue replicable on latest master. Seems to be happening when `SizedBox.shrink()` is added as a `WidgetSpan` along with `recognizer` property added to rest of the `TextSpans`.

Also happens with official RichText code sample from the [docs](https://api.flutter.dev/flutter/widgets/WidgetSpan-class.html) when we add `SizedBox.shrink()` and `tapRecognizer` to the sample code.

<details>
<summary>code sample with reproducible crash</summary>

```
Text.rich(
              TextSpan(
                children: <InlineSpan>[
                  TextSpan(text: 'Flutter is', recognizer: TapGestureRecognizer()..onTap = () {},),
                  WidgetSpan(
                      child: SizedBox.shrink(
                      //  width: 120,
                      //  height: 50,
                        child: Card(
                            child: Center(
                                child: Text('Hello World!')
                            )
                        ),
                      )
                  ),
                  TextSpan(text: 'the best!', recognizer: TapGestureRecognizer()..onTap = () {},),
                ],
              )
          )
```

</details>

<details>
<summary>flutter doctor -v</summary>

```
Flutter 1.22.0-10.0.pre.174 • channel master • https://github.com/flutter/flutter.git
Framework • revision 37b03ec2c2 (4 hours ago) • 2020-09-14 23:52:04 -0400
Engine • revision 7e962ba911
Tools • Dart 2.10.0 (build 2.10.0-124.0.dev)

Running flutter doctor...
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel master, 1.22.0-10.0.pre.174, on Mac OS X 10.15.4 19E2269, locale en-IN)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 11.5)
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.0)
[✓] VS Code (version 1.46.1)
[✓] Connected device (4 available)

• No issues found!

```

</details>
